libs/langchain-openai/src/chat_models.ts

```diff
@@ -843,4 +854,118 @@ export class ChatOpenAI<
       }
     );
   }
+
+  /**
+   * Model wrapper that returns outputs formatted to match the given schema.
+   *
+   * @template {any} RunInput The input type for the Runnable.
+   * @template {z.ZodObject<any, any, any, any>} RunOutput The output type for the Runnable, expected to be a Zod schema object for structured output validation.
+   * @template {RunnableConfig} CallOptions The type for call options, extending from RunnableConfig.
+   *
+   * @param {z.ZodEffects<RunOutput>} schema The schema for the structured output. Either as a ZOD schema or a valid JSON schema object.
+   * @param {string} name The name of the function to call.
+   * @param {"functionCalling" | "jsonMode"} method The method to use for getting the structured output. Defaults to "functionCalling".
+   * @param {boolean | undefined} includeRaw Whether to include the raw output in the result. Defaults to false.
+   * @returns {Runnable<RunInput, RunOutput, CallOptions>} A new runnable that calls the LLM with structured output.
+   */
+  withStructuredOutput<
```

Comment:
Are we not adding this to the core interface? Down not to for now, just checking

Reply:
We are, but not all of these args, py only had schema, i had to add name for JS